### PR TITLE
bugfix/19808-ordinal-rangeselector-states

### LINKED ISSUE #19808: Range selector week button weird behavior after using the scrollbar.
#### Expected behaviour
When clicking the week button, and scrolling with the scrollbar, the week button should remain active, as the range doesn't change.

#### Actual behaviour
When clicking the week button, and scrolling with the scrollbar, the week button doesn't remain active.

When I change the data range to a smaller one, at some point, the problem doesn't occur. Moreover, I noticed that even when I change the width of the chart, the result might change, see: 

https://github.com/highcharts/highcharts/assets/47400438/f999e071-8e98-4698-a5f2-cdee415d565f

When chart width is 800, everything works correctly, but when I change it to 700, we can see that the button doesn't remain active after scrolling. 

https://jsfiddle.net/BlackLabel/w9y1u8o0/

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/sp5fLr0g/

#### Product version
Highstock v11.1.0

#### Affected browser(s)
All

### PR DESCRIPTION
**Pull request by kamil-musialowski**
Fixed #19808, the range selector week button behaved weirdly after using the scrollbar.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205568209738908

### CODE REVIEW

File: samples/unit-tests/rangeselector/buttons-range-state/demo.js [hunk 1]
```diff
@@ -194,4 +185,41 @@ QUnit.test('Range selector buttons states (#3375)', function (assert) {
         chart.xAxis[0].dataMax - chart.xAxis[0].dataMin,
         'chart.fixedRange should be equal to maximum range if the clicked button is outside data range, #20327'
     );
+
+    // #19808
+    const ordinalData = [
+        [1706227200000, 2],    // 26.01
+        [1706592000000, 3],    // 30.01
+        [1706505600000, 1],    // 29.01
```
Comment (pawelfus) [inline samples/unit-tests/rangeselector/buttons-range-state/demo.js:193]: Unsorted data?

**Review by TorsteinHonsi — APPROVED** [review]
Awesome, thanks!

PS: @kamil-musialowski Should this PR be marked Ready to review?

**Review by pawelfus — CHANGES_REQUESTED** [review]

**Review by bm64 — APPROVED** [review]

**Review by pawelfus — APPROVED** [review]
Thanks!

**Review by pawellysy — APPROVED** [review]
LGTM

**Pull request merged**